acceptancetests/run-windows-job.sh

```diff
@@ -0,0 +1,45 @@
+#!/bin/bash
+#This script allows you to run arbitrary commands on windows slaves
+#You can also specify a tarball to be used as part of the payload
+#Usage: host (tarball or revision id or cmd_to_run)
+set -eux
+IFS='%'
+
+runcmd() {
+    #Make payload to run
+    payload="$ts.bat"
+    echo "cd C:\\Users\\Administrator\\payloads\\$ts" > $payload
+    echo "$1" >> $payload
+    #Ensure the error is returned if the command fails
+    echo "if errorlevel 1 (" >> $payload
+    echo "   exit /b %errorlevel%" >> $payload
+    echo ")" >> $payload
+    scp -i $JUJU_HOME/staging-juju-rsa -oStrictHostKeyChecking=no $payload $host:C:\\Users\\Administrator\\payloads\\
+    #Run test command via ssh
+    ssh $host C:\\Users\\Administrator\\payloads\\$payload
+}
+
+host=$1
+ts=$(date +%s)
+tarfile=''
+echo "Running on $host as payload $ts"
+runcmd "mkdir C:\\Users\\Administrator\\payloads\\$ts"
```

Comment:
Ohh.. I understand your meaning. I think I prefer putting all functions right at the top. But it does make it harder to read. Bash won't let you move things nicely because of how it's interpreted. That said, I think you could put the declarations above the function. I just don't want to split the code. I find that harder / more hidden to read.